### PR TITLE
[doctest] updating to version 1.2.0

### PR DESCRIPTION
**Pull request by onqtam**

### CODE REVIEW

File: cmake/projects/doctest/hunter.cmake [hunk 1]
```diff
@@ -14,16 +14,15 @@ hunter_add_version(
     PACKAGE_NAME
     doctest
     VERSION
-    "1.1.4-hunter-1"
+    "1.2.0"
```
Comment (ruslo) [inline cmake/projects/doctest/hunter.cmake]: Please add new `hunter_add_version` directive instead of removing old.

Comment (onqtam) [inline cmake/projects/doctest/hunter.cmake]: done

**Comment by ruslo** [comment]
Testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.1731
* https://travis-ci.org/ingenue/hunter/builds/233546102?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
> Testing

Some toolchains stop working. Was fine before:
* https://travis-ci.org/ingenue/hunter/builds/216186171

**Comment by ruslo** [comment]
`android-*` and `analyze` toolchains stop working. I guess Android lacking of C++11 support. Analyze report some issue, need to investigate is it something valid or not... Feel free to send new fixed version or just exclude toolchains from testing by sending pull request to [ingenue/hunter@pkg.doctest](https://github.com/ingenue/hunter/tree/pkg.doctest) branch. Pull request should contains only `.travis.yml`/`appveyor.yml` update, see example:
* https://github.com/ingenue/hunter/pull/56/files

**Comment by onqtam** [comment]
thats unfortunate... In version 1.2 all calls to ```printf``` or ```snprintf``` got prefixed because of [this PR](https://github.com/onqtam/doctest/pull/64/files) - and I include the C headers like ```<cheader>``` instead of ```<header.c>```...... so the android toolchain is at fault here.

And I analyze my code with clang-tidy-4 but I suppress some things - so I'm not sure what to do about the builds which fail now with clang-analyze - I feel such testing should be left up to the project itself

**Comment by onqtam** [comment]
I created a [PR](https://github.com/ingenue/hunter/pull/75) for that like you suggested

**Comment by ruslo** [comment]
Thanks.

Testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.1740
* https://travis-ci.org/ingenue/hunter/builds/233593160?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
Uploading:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.1741
* https://travis-ci.org/ingenue/hunter/builds/233600171?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
Released: https://github.com/ruslo/hunter/releases/tag/v0.18.62

**Pull request merged**